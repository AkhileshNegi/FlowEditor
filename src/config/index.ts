import ConfigProvider, { ConfigProviderContext } from './ConfigProvider';
import { Type, Mode, typeConfigList, getTypeConfig } from './typeConfigs';
import { Operator, operatorConfigList, getOperatorConfig } from './operatorConfigs';

export {
    ConfigProviderContext,
    Type,
    Mode,
    Operator,
    typeConfigList,
    getTypeConfig,
    operatorConfigList,
    getOperatorConfig
};

export default ConfigProvider;
