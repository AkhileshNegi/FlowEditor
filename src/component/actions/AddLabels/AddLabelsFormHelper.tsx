import { FormHelper, Types } from '../../../config/typeConfigs';
import { AddLabels } from '../../../flowTypes';
import { Asset, AssetType } from '../../../services/AssetService';
import { AddLabelsFormState } from '../../../store/nodeEditor';

export class AddLabelsFormHelper implements FormHelper {
    public actionToState(action: AddLabels): AddLabelsFormState {
        if (action) {
            return {
                type: action.type,
                labels: {
                    value: action.labels.map(label => {
                        return { id: label.uuid, name: label.name, type: AssetType.Label };
                    })
                },
                valid: true
            };
        }

        return {
            type: Types.add_input_labels,
            labels: { value: [] },
            valid: false
        };
    }

    public stateToAction(actionUUID: string, formState: AddLabelsFormState): AddLabels {
        return {
            labels: this.getAsset(formState.labels.value, AssetType.Label),
            type: formState.type,
            uuid: actionUUID
        };
    }

    private getAsset(assets: Asset[], type: AssetType): any[] {
        return assets.filter((asset: Asset) => asset.type === type).map((asset: Asset) => {
            return { uuid: asset.id, name: asset.name };
        });
    }
}
