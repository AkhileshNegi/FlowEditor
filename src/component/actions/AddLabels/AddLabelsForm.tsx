import { react as bindCallbacks } from 'auto-bind';
import * as React from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';

import { ConfigProviderContext } from '../../../config';
import { fakePropType } from '../../../config/ConfigProvider';
import { Types } from '../../../config/typeConfigs';
import { AddLabels, Label } from '../../../flowTypes';
import AssetService, { Asset, AssetType } from '../../../services/AssetService';
import { DispatchWithState } from '../../../store';
import { AddLabelsFunc, updateAddLabelsForm } from '../../../store/forms';
import { AddLabelsFormState } from '../../../store/nodeEditor';
import AppState from '../../../store/state';
import { validate, validateRequired } from '../../../store/validators';
import LabelsElement from '../../form/LabelsElement';
import { AddLabelsFormHelper } from './AddLabelsFormHelper';

export interface AddLabelsFormStoreProps {
    form: AddLabelsFormState;
    updateAddLabelsForm: AddLabelsFunc;
}

export interface AddLabelsFormPassedProps {
    action: AddLabels;
    updateAction: (action: AddLabels) => void;
    formHelper: AddLabelsFormHelper;
}

export type AddLabelsFormProps = AddLabelsFormPassedProps & AddLabelsFormStoreProps;

export const mapLabelsToAssets = (labels: Label[]): Asset[] =>
    labels.map(({ name, uuid }) => ({ name, id: uuid, type: AssetType.Label }));

export const mapAssetsToLabels = (searchResults: Asset[]): Label[] =>
    searchResults.map(({ id, name }) => ({ uuid: id, name }));

export const createNewAddLabelAction = ({ uuid }: AddLabels, labels: Asset[]) => ({
    uuid,
    type: Types.add_input_labels,
    labels: mapAssetsToLabels(labels)
});

export const LABEL = 'Select the label(s) to apply to the incoming message.';
export const PLACEHOLDER = 'Enter the name of an existing label or create a new one';

export const controlLabelSpecId = 'label';

export class AddLabelsForm extends React.PureComponent<AddLabelsFormProps> {
    public static contextTypes = {
        assetService: fakePropType
    };

    constructor(props: AddLabelsFormProps, context: ConfigProviderContext) {
        super(props);

        bindCallbacks(this, {
            include: [/^on/, /^handle/]
        });
    }

    public onValid(): void {
        const newAction = this.props.formHelper.stateToAction(
            this.props.action.uuid,
            this.props.form
        );
        this.props.updateAction(newAction);
    }

    public validate(): boolean {
        return this.handleLabelChange(this.props.form.labels.value);
    }

    public handleLabelChange(selected: Asset[]): boolean {
        return (this.props.updateAddLabelsForm({
            labels: validate('Labels', selected, [validateRequired])
        }) as any).valid;
    }

    private getLabels(): Asset[] {
        if (!this.props.action.labels.length) {
            return [];
        }
        return mapLabelsToAssets(this.props.action.labels);
    }

    public render(): JSX.Element {
        return (
            <>
                <p data-spec={controlLabelSpecId}>{LABEL}</p>
                <LabelsElement
                    name="Labels"
                    placeholder={PLACEHOLDER}
                    assets={this.context.assetService.getLabelAssets()}
                    entry={this.props.form.labels}
                    onChange={this.handleLabelChange}
                />
            </>
        );
    }
}

/* istanbul ignore next */
const mapStateToProps = ({ nodeEditor: { form } }: AppState) => ({
    form
});

/* istanbul ignore next */
const mapDispatchToProps = (dispatch: DispatchWithState) =>
    bindActionCreators({ updateAddLabelsForm }, dispatch);

const ConnectedAddLabelsForm = connect(mapStateToProps, mapDispatchToProps, null, {
    withRef: true
})(AddLabelsForm);

export default ConnectedAddLabelsForm;
