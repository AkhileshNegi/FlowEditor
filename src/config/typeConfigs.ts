import AddLabelsComp from '../component/actions/AddLabels/AddLabels';
import AddLabelsForm from '../component/actions/AddLabels/AddLabelsForm';
import CallWebhookComp from '../component/actions/CallWebhook/CallWebhook';
import AddGroupsForm from '../component/actions/ChangeGroups/AddGroupsForm';
import ChangeGroupsComp from '../component/actions/ChangeGroups/ChangeGroups';
import RemoveGroupsForm from '../component/actions/ChangeGroups/RemoveGroupsForm';
import MissingComp from '../component/actions/Missing/Missing';
import SendBroadcastComp from '../component/actions/SendBroadcast/SendBroadcast';
import SendBroadcastForm from '../component/actions/SendBroadcast/SendBroadcastForm';
import { SendBroadcastFormHelper } from '../component/actions/SendBroadcast/SendBroadcastFormHelper';
import SendEmailComp from '../component/actions/SendEmail/SendEmail';
import SendEmailForm from '../component/actions/SendEmail/SendEmailForm';
import SendMsgComp from '../component/actions/SendMsg/SendMsg';
import SendMsgForm from '../component/actions/SendMsg/SendMsgForm';
import SetContactAttrib from '../component/actions/SetContactAttrib/SetContactAttrib';
import SetContactAttribForm from '../component/actions/SetContactAttrib/SetContactAttribForm';
import { SetContactAttribFormHelper } from '../component/actions/SetContactAttrib/SetContactAttribFormHelper';
import SetRunResultComp from '../component/actions/SetRunResult/SetRunResult';
import SetRunResultForm from '../component/actions/SetRunResult/SetRunResultForm';
import StartFlowComp from '../component/actions/StartFlow/StartFlow';
import StartSessionComp from '../component/actions/StartSession/StartSession';
import StartSessionForm from '../component/actions/StartSession/StartSessionForm';
import { StartSessionFormHelper } from '../component/actions/StartSession/StartSessionFormHelper';
import GroupsRouter from '../component/routers/GroupsRouter';
import SubflowRouter from '../component/routers/SubflowRouter';
import SwitchRouter from '../component/routers/SwitchRouter';
import WebhookRouter from '../component/routers/WebhookRouter';
import { AnyAction, RouterTypes, UINodeTypes } from '../flowTypes';
import { NodeEditorForm } from '../store/nodeEditor';

/*
Old name	                New name	                Event(s) generated

add_urn	                    add_contact_urn	            contact_urn_added
add_to_group	            add_contact_groups	        contact_groups_added
remove_from_group	        remove_contact_groups	    contact_groups_removed
set_preferred_channel	    set_contact_channel	        contact_channel_changed
update_contact	            set_contact_name	        contact_name_changed
update_contact	            set_contact_language	    contact_language_changed
update_contact	            set_contact_timezone	    contact_timezone_changed
save_contact_field      	set_contact_field	        contact_field_changed
save_flow_result	        set_run_result	            run_result_changed
call_webhook	            call_webhook	            webhook_called
add_label	                add_input_labels	        input_labels_added
reply	                    send_msg	                msg_created
send_email	                send_email	                email_created / email_sent
send_msg	                send_broadcast	            broadcast_created
start_flow	                start_flow	                flow_triggered
start_session	            start_session	            session_triggered
*/

export const enum Types {
    add_contact_urn = 'add_contact_urn',
    add_contact_groups = 'add_contact_groups',
    add_input_labels = 'add_input_labels',
    remove_contact_groups = 'remove_contact_groups',
    set_contact_channel = 'set_contact_channel',
    set_contact_name = 'set_contact_name',
    set_contact_field = 'set_contact_field',
    set_run_result = 'set_run_result',
    call_webhook = 'call_webhook',
    send_msg = 'send_msg',
    send_email = 'send_email',
    send_broadcast = 'send_broadcast',
    start_flow = 'start_flow',
    start_session = 'start_session',
    split_by_expression = 'split_by_expression',
    split_by_groups = 'split_by_groups',
    wait_for_response = 'wait_for_response',

    missing = 'missing'
}

export enum Mode {
    EDITING = 0x1,
    TRANSLATING = 0x2,
    ALL = EDITING | TRANSLATING
}

export interface FormHelper {
    actionToState: (action: AnyAction, actionType?: Types) => NodeEditorForm;
    stateToAction: (actionUUID: string, formState: NodeEditorForm, formType?: Types) => AnyAction;
}

export interface Type {
    type: Types;
    name: string;
    description: string;
    allows(mode: Mode): boolean;
    component?: React.SFC<AnyAction>;
    form?: React.ComponentClass<any>;
    formHelper?: FormHelper;
    advanced?: Mode;
    aliases?: string[];
}

export interface TypeMap {
    [propName: string]: Type;
}

export type GetTypeConfig = (type: string) => Type;

export function allows(mode: Mode): boolean {
    return (this.advanced & mode) === mode;
}

const ContactAttribHelper = new SetContactAttribFormHelper();

export const typeConfigList: Type[] = [
    /** Actions */
    {
        type: Types.missing,
        name: 'Missing',
        description: ' ** Unsupported ** ',
        component: MissingComp,
        allows
    },
    {
        type: Types.send_msg,
        name: 'Send Message',
        description: 'Send the contact a message',
        form: SendMsgForm,
        component: SendMsgComp,
        advanced: Mode.EDITING,
        allows
    },
    {
        type: Types.send_broadcast,
        name: 'Send Broadcast',
        description: 'Send somebody else a message',
        form: SendBroadcastForm,
        formHelper: new SendBroadcastFormHelper(),
        component: SendBroadcastComp,
        allows
    },
    {
        type: Types.add_input_labels,
        name: 'Add Labels',
        description: 'Label the incoming message',
        form: AddLabelsForm,
        component: AddLabelsComp,
        allows
    },
    {
        type: Types.add_contact_groups,
        name: 'Add to Group',
        description: 'Add the contact to a group',
        form: AddGroupsForm,
        component: ChangeGroupsComp,
        allows
    },
    {
        type: Types.remove_contact_groups,
        name: 'Remove from Group',
        description: 'Remove the contact from a group',
        form: RemoveGroupsForm,
        component: ChangeGroupsComp,
        allows
    },
    {
        type: Types.set_contact_field,
        name: 'Update Contact',
        description: 'Update the contact',
        form: SetContactAttribForm,
        formHelper: ContactAttribHelper,
        component: SetContactAttrib,
        allows
    },
    {
        type: Types.set_contact_name,
        name: 'Update Contact',
        description: 'Update the contact',
        form: SetContactAttribForm,
        formHelper: ContactAttribHelper,
        component: SetContactAttrib,
        allows
    },
    {
        type: Types.send_email,
        name: 'Send Email',
        description: 'Send an email',
        form: SendEmailForm,
        component: SendEmailComp,
        allows
    },
    {
        type: Types.set_run_result,
        name: 'Save Flow Result',
        description: 'Save a result for this flow',
        form: SetRunResultForm,
        component: SetRunResultComp,
        allows
    },
    // {type: 'set_preferred_channel', name: 'Set Preferred Channel', description: 'Set their preferred channel', component: Missing},
    /** Hybrids */
    {
        type: Types.call_webhook,
        name: 'Call Webhook',
        description: 'Call a webook',
        form: WebhookRouter,
        component: CallWebhookComp,
        advanced: Mode.EDITING,
        aliases: [UINodeTypes.webhook],
        allows
    },
    {
        type: Types.start_flow,
        name: 'Start a Flow',
        description: 'Enter another flow',
        form: SubflowRouter,
        component: StartFlowComp,
        aliases: [UINodeTypes.subflow],
        allows
    },
    {
        type: Types.start_session,
        name: 'Start Somebody Else',
        description: 'Start somebody else in a flow',
        form: StartSessionForm,
        formHelper: new StartSessionFormHelper(),
        component: StartSessionComp,
        allows
    },

    /** Routers */
    {
        type: Types.split_by_expression,
        name: 'Split by Expression',
        description: 'Split by a custom expression',
        form: SwitchRouter,
        advanced: Mode.TRANSLATING,
        allows
    },
    {
        type: Types.split_by_groups,
        name: 'Split by Group Membership',
        description: 'Split by group membership',
        form: GroupsRouter,
        allows
    },
    {
        type: Types.wait_for_response,
        name: 'Wait for Response',
        description: 'Wait for the contact to respond',
        form: SwitchRouter,
        advanced: Mode.TRANSLATING,
        aliases: [RouterTypes.switch],
        allows
    }
    // {type: 'random', name: 'Random Split', description: 'Split them up randomly', form: RandomRouterForm}
];

export const typeConfigMap: TypeMap = typeConfigList.reduce((map: TypeMap, typeConfig: Type) => {
    map[typeConfig.type] = typeConfig;
    if (typeConfig.aliases) {
        typeConfig.aliases.forEach((alias: string) => (map[alias] = typeConfig));
    }
    return map;
}, {});

/**
 * Shortcut for constant lookup of type config in type configs map
 * @param {string} type - The type of the type config to return, e.g. 'send_msg'
 * @returns {Object} - The type config found at typeConfigs[type] or -1
 */
export const getTypeConfig = (type: Types | RouterTypes | UINodeTypes): Type => {
    let actionConfig = typeConfigMap[type];

    if (!actionConfig) {
        actionConfig = typeConfigMap.missing;
    }
    return actionConfig;
};
