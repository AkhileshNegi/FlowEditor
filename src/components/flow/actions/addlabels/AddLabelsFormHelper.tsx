import { FormHelper, Types } from '~/config/typeConfigs';
import { AddLabels } from '~/flowTypes';
import { Asset, AssetType } from '~/services/AssetService';
import { AddLabelsFormState, NodeEditorSettings } from '~/store/nodeEditor';

export class AddLabelsFormHelper implements FormHelper {
    public initializeForm(settings: NodeEditorSettings): AddLabelsFormState {
        if (settings.originalAction && settings.originalAction.type === Types.add_input_labels) {
            const action = settings.originalAction as AddLabels;
            return {
                labels: {
                    value: action.labels.map(label => {
                        return { id: label.uuid, name: label.name, type: AssetType.Label };
                    })
                },
                valid: true
            };
        }

        return {
            labels: { value: [] },
            valid: false
        };
    }

    public stateToAction(actionUUID: string, formState: AddLabelsFormState): AddLabels {
        return {
            type: Types.add_input_labels,
            labels: this.getAsset(formState.labels.value, AssetType.Label),
            uuid: actionUUID
        };
    }

    private getAsset(assets: Asset[], type: AssetType): any[] {
        return assets.filter((asset: Asset) => asset.type === type).map((asset: Asset) => {
            return { uuid: asset.id, name: asset.name };
        });
    }
}
