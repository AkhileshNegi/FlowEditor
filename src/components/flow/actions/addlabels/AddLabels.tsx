import * as React from 'react';
import { renderAssetList } from 'components/flow/actions/helpers';
import { fakePropType } from 'config/ConfigProvider';
import { AddLabels, MissingDependencies, Dependency } from 'flowTypes';
import { AssetType } from 'store/flowContext';

export const MAX_TO_SHOW = 5;

const AddLabelsComp: React.SFC<AddLabels & MissingDependencies> = (
  { labels, missingDependencies },
  context: any
): JSX.Element => {
  return (
    <>
      {renderAssetList(
        labels.map(label => {
          return {
            id: label.uuid,
            name: label.name,
            type: AssetType.Label,
            missing: !!missingDependencies.find((dep: Dependency) => dep.uuid === label.uuid)
          };
        }),
        MAX_TO_SHOW,
        context.config.endpoints
      )}
    </>
  );
};

AddLabelsComp.contextTypes = {
  config: fakePropType
};

export default AddLabelsComp;
