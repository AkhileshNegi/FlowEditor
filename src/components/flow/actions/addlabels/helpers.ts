import { AddLabelsFormState } from '~/components/flow/actions/addlabels/AddLabelsForm';
import { getActionUUID } from '~/components/flow/actions/helpers';
import { Types } from '~/config/interfaces';
import { AddLabels } from '~/flowTypes';
import { Asset, AssetType } from '~/store/flowContext';
import { NodeEditorSettings } from '~/store/nodeEditor';

export const initializeForm = (settings: NodeEditorSettings): AddLabelsFormState => {
    if (settings.originalAction && settings.originalAction.type === Types.add_input_labels) {
        const action = settings.originalAction as AddLabels;
        return {
            labels: {
                value: action.labels.map(label => {
                    return { id: label.uuid, name: label.name, type: AssetType.Label };
                })
            },
            valid: true
        };
    }

    return {
        labels: { value: [] },
        valid: false
    };
};

export const stateToAction = (
    settings: NodeEditorSettings,
    formState: AddLabelsFormState
): AddLabels => {
    return {
        type: Types.add_input_labels,
        labels: this.getAsset(formState.labels.value, AssetType.Label),
        uuid: getActionUUID(settings, Types.add_input_labels)
    };
};

export const getAsset = (assets: Asset[], type: AssetType): any[] => {
    return assets.filter((asset: Asset) => asset.type === type).map((asset: Asset) => {
        return { uuid: asset.id, name: asset.name };
    });
};
