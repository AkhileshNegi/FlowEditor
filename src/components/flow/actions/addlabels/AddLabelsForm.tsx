import { react as bindCallbacks } from 'auto-bind';
import Dialog, { ButtonSet } from 'components/dialog/Dialog';
import { ActionFormProps } from 'components/flow/props';
import AssetSelector from 'components/form/assetselector/AssetSelector';
import TypeList from 'components/nodeeditor/TypeList';
import { fakePropType } from 'config/ConfigProvider';
import * as React from 'react';
import { Asset } from 'store/flowContext';
import { AssetArrayEntry, FormState, mergeForm } from 'store/nodeEditor';
import { Required, validate } from 'store/validators';

import { initializeForm, stateToAction } from './helpers';

export interface AddLabelsFormState extends FormState {
  labels: AssetArrayEntry;
}

export const controlLabelSpecId = 'label';

export default class AddLabelsForm extends React.PureComponent<
  ActionFormProps,
  AddLabelsFormState
> {
  public static contextTypes = {
    assetService: fakePropType
  };

  constructor(props: ActionFormProps) {
    super(props);

    this.state = initializeForm(this.props.nodeSettings);
    bindCallbacks(this, {
      include: [/^on/, /^handle/]
    });
  }

  public handleSave(): void {
    const valid = this.handleLabelsChanged(this.state.labels.value!);

    if (valid) {
      const newAction = stateToAction(this.props.nodeSettings, this.state);
      this.props.updateAction(newAction);
      this.props.onClose(false);
    }
  }

  public handleLabelsChanged(selected: Asset[]): boolean {
    const updates: Partial<AddLabelsFormState> = {
      labels: validate('Labels', selected, [Required])
    };

    const updated = mergeForm(this.state, updates);
    this.setState(updated);
    return updated.valid;
  }

  private getButtons(): ButtonSet {
    return {
      primary: { name: 'Ok', onClick: this.handleSave },
      secondary: { name: 'Cancel', onClick: () => this.props.onClose(true) }
    };
  }

  public handleCreateAssetFromInput(input: string): any {
    return { name: input };
  }

  public handleLabelCreated(label: Asset): void {
    // update our store with our new group
    this.props.addAsset('labels', label);

    this.handleLabelsChanged(this.state.labels.value!.concat(label));
  }

  public render(): JSX.Element {
    const typeConfig = this.props.typeConfig;
    return (
      <Dialog title={typeConfig.name} headerClass={typeConfig.type} buttons={this.getButtons()}>
        <TypeList __className="" initialType={typeConfig} onChange={this.props.onTypeChange} />
        <p data-spec={controlLabelSpecId}>Select the labels to apply to the incoming message.</p>

        <AssetSelector
          name="Labels"
          placeholder="Enter the name of an existing label or create a new one"
          assets={this.props.assetStore.labels}
          entry={this.state.labels}
          searchable={true}
          multi={true}
          onChange={this.handleLabelsChanged}
          createPrefix="Create Label: "
          createAssetFromInput={this.handleCreateAssetFromInput}
          onAssetCreated={this.handleLabelCreated}
        />
      </Dialog>
    );
  }
}
