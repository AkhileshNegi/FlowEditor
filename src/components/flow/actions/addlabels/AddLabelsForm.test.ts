import AddLabelsForm, {
    AddLabelsFormProps
} from '~/components/flow/actions/addlabels/AddLabelsForm';
import { AddLabelsFormHelper } from '~/components/flow/actions/addlabels/AddLabelsFormHelper';
import { getTypeConfig, Types } from '~/config/typeConfigs';
import { Label } from '~/flowTypes';
import { composeComponentTestUtils } from '~/testUtils';
import { createAddLabelsAction } from '~/testUtils/assetCreators';

const { assets: labels } = require('~/test/assets/labels.json') as {
    assets: Label[];
};

const formHelper = new AddLabelsFormHelper();
const action = createAddLabelsAction(labels);
const sendConfig = getTypeConfig(Types.send_broadcast);

const baseProps: AddLabelsFormProps = {
    updateAction: jest.fn(),
    onTypeChange: jest.fn(),
    onClose: jest.fn(),
    nodeSettings: {
        originalNode: null,
        originalAction: action
    },
    typeConfig: sendConfig,
    formHelper
};

const { setup, spyOn } = composeComponentTestUtils(AddLabelsForm, baseProps);

describe(AddLabelsForm.name, () => {
    describe('render', () => {
        it('should pass LabelsElement labels if they exist on the action', () => {
            const { wrapper } = setup();
            expect(wrapper).toMatchSnapshot();
        });
    });
});
