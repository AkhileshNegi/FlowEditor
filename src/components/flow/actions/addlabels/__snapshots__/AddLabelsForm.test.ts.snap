// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddLabelsForm render should pass LabelsElement labels if they exist on the action 1`] = `
<Dialog
  buttons={
    Object {
      "primary": Object {
        "name": "Ok",
        "onClick": [Function],
      },
      "secondary": Object {
        "name": "Cancel",
        "onClick": [Function],
      },
    }
  }
  headerClass="send_broadcast"
  title="Send Broadcast"
>
  <TypeList
    __className=""
    initialType={
      Object {
        "component": [Function],
        "description": "Send somebody else a message",
        "form": [Function],
        "formHelper": SendBroadcastFormHelper {},
        "name": "Send Broadcast",
        "type": "send_broadcast",
      }
    }
    onChange={[MockFunction]}
  />
  <p
    data-spec="label"
  >
    Select the label(s) to apply to the incoming message.
  </p>
  <LabelsElement
    assets={
      LabelAssets {
        "assetType": "label",
        "assets": Object {},
        "endpoint": "/assets/labels.json",
        "idProperty": "uuid",
        "localStorage": true,
      }
    }
    entry={
      Object {
        "value": Array [
          Object {
            "id": "232b99d1-579f-4d7f-b750-4b034927e9f8",
            "name": "Help",
            "type": "label",
          },
          Object {
            "id": "ab58dffa-98da-40ba-bebc-489ec6ff7170",
            "name": "Feedback",
            "type": "label",
          },
        ],
      }
    }
    name="Labels"
    onChange={[Function]}
    placeholder="Enter the name of an existing label or create a new one"
    required={true}
    searchPromptText="Invalid label"
  />
</Dialog>
`;
